File: go/aoc_2021/internal/day01/day01.go
package day01

import (
	"fmt"
	"gitlab.com/danscott/adventofcode/go/common/inputs"
	"gitlab.com/danscott/adventofcode/go/common/runner"
)

func New() runner.Day {
	return &day01{}
}

type day01 struct {
	ms []int
}

func (d *day01) Open() {
	d.ms = inputs.LinesAsInt(2021, 1)
}

func (d *day01) Close() {
	d.ms = nil
}

func (d *day01) Part1() string {
	ct := 0
	for i, m := range d.ms[1:] {
		if m > d.ms[i] {
			ct++
		}
	}

	return fmt.Sprint(ct)

}

func (d *day01) Part2() string {
	ct := 0
	for i := 3; i < len(d.ms); i++ {
		if d.sumAt(i) > d.sumAt(i-1) {
			ct++
		}
	}
	return fmt.Sprint(ct)
}

func (d *day01) sumAt(i int) int {
	return d.ms[i-2] + d.ms[i-1] + d.ms[i]
}

File: dotnet/AdventOfCode.2021/Day01/Day01.cs
using AdventOfCode.Base;

namespace AdventOfCode._2021.Day01;

public class Day01 : IDay
{
    private List<int>? _readings;

    private List<int> Readings => _readings ?? new List<int>();
    
    public void Dispose()
    {
        _readings = null;
    }

    public int Number => 1;
    public void Open()
    {
        _readings = Inputs.LinesAsInt(2021, 1).ToList();
    }

    public string Part1() => Reading.ScanInput(Readings).Count(r => r.HasIncreased).ToString();

    public string Part2() => Reading.ScanWindows(Readings).Count(r => r.HasWindowIncreased).ToString();

}